pex/interpreter.py

```diff
@@ -357,8 +390,8 @@ def _resolve_pyvenv_canonical_python_binary(
         if not os.path.islink(maybe_venv_python_binary):
             return None
 
-        home_dir = cls._find_pyvenv_home(maybe_venv_python_binary)
-        if os.path.dirname(real_binary) != home_dir:
+        pyvenv_cfg = cls._find_pyvenv_cfg(maybe_venv_python_binary)
+        if pyvenv_cfg is None:
```

Comment:
The sanity check on the LHS got in the way of detecting both venvs created by virtualenv (virtualenv records home as `sys.prefix`, not `sys.prefix`/bin like pyvenv / PEP-405) and dealing with nested venvs. The unit test exercises the latter. A follow-on that adds the `Venv` `PEX_TOOL` will exercise the former since it embeds a copy of old virtualenv to support Python 2.7.